pkg/api/vlabs/types.go

```diff
@@ -195,10 +195,12 @@ type AgentPoolProfile struct {
 	DiskSizesGB         []int  `json:"diskSizesGB,omitempty"`
 	VnetSubnetID        string `json:"vnetSubnetID,omitempty"`
 	IPAddressCount      int    `json:"ipAddressCount,omitempty"`
-
+	OsImageOffer      	string `json:"osImageOffer,omitempty"`
```

Comment:
- ARM uses the same model so i guess it is OK to follow the same approach. Intention is to first introduce change for agent nodes and let it bake before changing the masters. When we have Ignition support we can move master and these settings will go to linux profile otherwise it will be exactly same as using defaults.  Currently it is hard coded to use Image in variables but ultimately we should be allowing users to customise it from the acs-engine itself.

- Are there any issues for mixed mode cluster (CoreOS  Agent + Ubuntu Masters). My understanding is this is fine and supported. 

- Please point me to existing tests specific to Ubuntu. I will do the same for CoreOS.